pennylane/circuit_drawer.py

```diff
@@ -0,0 +1,916 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the CircuitDrawer class which is used to draw CircuitGraph instances.
+"""
+import abc
+import math
+from collections import OrderedDict
+
+import numpy as np
+
+import pennylane as qml
+
+# pylint: disable=too-many-branches,too-many-arguments,too-many-return-statements,too-many-statements,consider-using-enumerate,too-many-instance-attributes
+
+
+def _transpose(target_list):
```

Comment:
This method seems to work in such a way, that some sublists might be truncated during transposing:
```python
In [1]: a = _transpose([[1,2], [3, 5, 7]])                                                                                                                                                                  

In [2]: a                                                                                                                                                                                                   
Out[2]: [[1, 3], [2, 5]]
```
This behaviour could be included here.

Also, giving the example of transposing a matrix might be useful (when reading for the first time, I was unsure if this would be the same thing, though it might just be me :sweat_smile: )

Reply:
Actually, this whole `Grid` class is somewhat "bad" in the sense the I just want to have a 2D array. I don't remember why I did not use `ndarray(object)`. I guess there was a problem when you wanted to edit strings inside a numpy array.

I think I will refactor this more toroughly